events/sinks/gcl/gcl.go

```diff
@@ -80,21 +79,21 @@ func (sink *gclSink) Stop() {
 }
 
 func CreateGCLSink(uri *url.URL) (core.EventSink, error) {
-	if err := gce_util.EnsureOnGCE(); err != nil {
-		return nil, err
-	}
-
-	// Detect project ID
-	projectId, err := gce.ProjectID()
+	client, err := google.DefaultClient(oauth2.NoContext, gcl.LoggingWriteScope)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating oauth2 client: %v", err)
 	}
 
 	// Create Google Cloud Logging service.
-	client := oauth2.NewClient(oauth2.NoContext, google.ComputeTokenSource(""))
 	gclService, err := gcl.New(client)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error creating GCM service: %v", err)
+	}
+
+	// Detect project ID
+	projectId, err := gce_util.GetProjectId()
```

Comment:
Does this work on AWS? What does it return? Can you explain it somewhere, preferably in a comment for GetProjectId() function?

Reply:
Yes, it does. In the use case I demonstrated in the PR description, project info is pulled out of the environment (GOOGLE_APPLICATION_CREDENTIALS). This is core heapster code: https://github.com/kubernetes/heapster/blob/974094fed3de505402451b6282859ce91bf88691/common/gce/gce.go#L45